### PR TITLE
Add inquiry vue-recaptcha to Inquiry form

### PR DESCRIPTION
**Pull request by B77Mills**

### CODE REVIEW

File: packages/marko-web-inquiry/env.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ const { cleanEnv, validators, str } = require('@base-cms/env');
 const { nonemptystr } = validators;
 
 module.exports = cleanEnv(process.env, {
+  // these are the google recaptcha test keys
+  RECAPTCHA_SITE_KEY: str({ desc: 'An site key for sending recaptcha validation.', default: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI' }),
+  RECAPTCHA_SECRET_KEY: str({ desc: 'A secret key for sending recaptcha validation.', default: '6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe' }),
```
Comment (solocommand) [inline packages/marko-web-inquiry/env.js:8]: @B77Mills The secret key cannot include a default value. Also, this key must now be immediately invalidated and new keys rotated in wherever it was used, as this is a public commit.

Comment (B77Mills) [reply packages/marko-web-inquiry/env.js:8]: @solocommand  These are the google dev keys

File: packages/marko-web-inquiry/env.js [hunk 2]
```diff
@@ -3,6 +3,9 @@ const { cleanEnv, validators, str } = require('@base-cms/env');
 const { nonemptystr } = validators;
 
 module.exports = cleanEnv(process.env, {
+  // these are the google recaptcha test keys
+  RECAPTCHA_SITE_KEY: str({ desc: 'An site key for sending recaptcha validation.', devDefault: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI' }),
+  RECAPTCHA_SECRET_KEY: str({ desc: 'A secret key for sending recaptcha validation.', devDefault: '6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe' }),
```
Comment (solocommand) [inline packages/marko-web-inquiry/env.js:8]: This should use the `nonemptystr` validator instead of `str`

**Comment by B77Mills** [comment]
@solocommand If you could re-review this it wold be great.  I made some logic changes to how some of it works.  I tested it on the sample LFW site and it passes all of the reCaptcha stuff, but fails on radix.  I also added the default RECAPTCHA_SITE_KEY in rancher for the websites.  

Thanks
Brian

File: packages/marko-web-inquiry/browser/default-form.vue [hunk 3]
```diff
@@ -156,6 +165,10 @@ export default {
       type: String,
       default: null,
     },
+    sitekey: {
+      type: String,
+      default: null,
```
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:170]:
```suggestion
      required: true,
```

Make the sitekey required, as you're using it below.

File: yarn.lock [hunk 4]
```diff
@@ -16093,6 +16093,11 @@ vue-loader@^15.8.3:
     vue-hot-reload-api "^2.3.0"
     vue-style-loader "^4.1.0"
 
+vue-recaptcha@^1.3.0:
```
Comment (solocommand) [inline yarn.lock:16096]: For the sake of consistency, use the same vue-recaptcha version we use elsewhere (Check EBM website contact-us package) (`1.2.0`), or upgrade both versions. The only major change I see between 1.2 and 1.3 is the addition of language support, and there don't appear to be any backwards incompatible changes.

File: packages/marko-web-inquiry/browser/default-form.vue [hunk 5]
```diff
@@ -131,6 +131,14 @@
...
+      ref="invisibleRecaptcha"
+      size="invisible"
+      :sitekey="sitekey"
+      :load-recaptcha-script="true"
+      @verify="onVerify"
+      @expired="onExpired"
+    />
     <button type="submit" class="btn btn-primary" :disabled="loading">
       Submit
     </button>
```
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:144]: You could simplify a lot of this logic by wrapping the submit button inside the `vue-recaptcha` component itself, as done in the [contact-us form](https://github.com/base-cms-websites/endeavor-business-media/blob/master/packages/contact-us/browser/form.vue#L93). Unless there were major changes in `v1.3.0` of the library, you shouldn't need the `size` or `load-...` params.

This also has the added benefit of allowing you to drop your `onSubmit` method, which is calling an undocumented method in the vue-recaptcha component 😄

File: packages/marko-web-inquiry/browser/default-form.vue [hunk 6]
```diff
@@ -169,7 +182,17 @@ export default {
     comments: '',
   }),
   methods: {
-    async submit() {
+    onSubmit() {
+      this.$refs.invisibleRecaptcha.execute();
+    },
+    onVerify(response) {
+      this.submit(response);
+    },
```
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:190]: No need to create a method to call another method -- make the `verify` argument passed to the component your `submit` method

File: packages/marko-web-inquiry/routers/submit.js [hunk 7]
```diff
@@ -33,6 +56,7 @@ module.exports = ({ queryFragment, notification, confirmation }) => asyncRoute(a
       payload,
       addresses,
     }),
+    validateRecaptcha(payload),
```
Comment (solocommand) [inline packages/marko-web-inquiry/routers/submit.js:59]: ❗ Validation should happen before the inquiry is stored

File: yarn.lock [hunk 8]
```diff
@@ -16093,7 +16093,7 @@ vue-loader@^15.8.3:
     vue-hot-reload-api "^2.3.0"
     vue-style-loader "^4.1.0"
 
-vue-recaptcha@^1.3.0:
+vue-recaptcha@^1.2.0:
   version "1.3.0"
```
Comment (solocommand) [inline yarn.lock]: This still has v`1.3.0` installed -- re-run yarn to update the lockfile.

File: packages/marko-web-inquiry/routers/submit.js [hunk 9]
```diff
@@ -49,14 +49,15 @@ module.exports = ({ queryFragment, notification, confirmation }) => asyncRoute(a
   };
 
   await Promise.all([
+    // validate google reCaptcha
+    await validateRecaptcha(payload),
     // Store the submission
     storeInquiry({
```
Comment (solocommand) [inline packages/marko-web-inquiry/routers/submit.js:55]: Unless you explicitly want to store the submission, you need to wait for the validation to pass _before_ storing it. `Promise.all` will execute all promises at the same time -- moving it up in the list won't cause it to wait.
```suggestion
  // validate google reCaptcha
  await validateRecaptcha(payload);

  await Promise.all([
    // Store the submission
    storeInquiry({
```

**Pull request merged**